lib/beaker/host/unix/exec.rb

```diff
@@ -142,7 +142,7 @@ def delete_env_var key, val
   #  host.get_env_var('path')
   def get_env_var key
     key = key.to_s
-    exec(Beaker::Command.new("env | grep #{key}"), :accept_all_exit_codes => true).stdout.chomp
+    exec(Beaker::Command.new("env | grep ^#{key}="), :accept_all_exit_codes => true).stdout.chomp
```

Comment:
For @ferglor & @glennsarti:
I agree with @glennsarti that it could definitely be better, but is outside the scope of this PR.

For @rick:
This is wide open, but only leaves a particular SUT open to arbitrary command execution, since it is a host command. Since for the most part SUTs are virtualized, no one tends to care about hosing one very much.

With PR testing, I could see someone submitting code to demolish a SUT, but the effect of that is minimal. If we were to hunt down a command that runs on the beaker coordinator itself, I could see that doing potentially much more damage in today's world, since our Jenkins slaves tend to be static hosts. Note that I'm not pretending that any of those commands are secured, I don't think this would be a difficult task.

My recommendation here is to have a whitelist like beaker does on when Jenkins will run your acceptance tests, and use discretion before signalling that it's OK for Jenkins to pick up new code to run.

Also in the next iteration of CI, the Jenkins slaves themselves will by dynamic and throw-away. Since that's the case, and that's how we would recommend someone setup their Jenkins to run beaker testing, this danger will be mitigated soon.